testing/upload_validation.ts

```diff
@@ -480,18 +480,126 @@ const booleanPropertySchema = zodCompleteObject<BooleanSchema & ObjectSchemaProp
   ...basePropertyValidators,
 });
 
-const numberPropertySchema = zodCompleteObject<NumberSchema & ObjectSchemaProperty>({
+const baseNumberPropertySchema = zodCompleteStrictObject<BaseNumberSchema & ObjectSchemaProperty>({
+  type: zodDiscriminant(ValueType.Number),
+  ...basePropertyValidators,
+})
+
+const numericPropertySchema = baseNumberPropertySchema.extend({
+  codaType: zodDiscriminant(ValueHintType.Percent).optional(),
+  precision: z.number().optional(),
+  useThousandsSeparator: z.boolean().optional(),
+})
+
+const scalePropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Scale),
+  maximum: z.number().optional(),
+  icon: z.nativeEnum(ScaleIconSet).optional(),
+  ...basePropertyValidators,
+})
+
+const sliderPropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Slider),
+  maximum: z.number().optional(),
+  minimum: z.number().optional(),
+  step: z.number().optional(),
+  ...basePropertyValidators,
+})
+
+const currencyPropertySchema = baseNumberPropertySchema.extend({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Currency),
+  precision: z.number().optional(),
+  currencyCode: z.string().optional(),
+  format: z.nativeEnum(CurrencyFormat).optional(),
+  ...basePropertyValidators,
+})
+
+const numericDatePropertySchema = zodCompleteObject<NumericDateSchema & ObjectSchemaProperty>({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.Date),
+  format: z.string().optional(),
+  ...basePropertyValidators,
+})
+
+const numericTimePropertySchema = zodCompleteObject<NumericTimeSchema & ObjectSchemaProperty>({
   type: zodDiscriminant(ValueType.Number),
-  codaType: z.enum([...NumberHintValueTypes]).optional(),
+  codaType: zodDiscriminant(ValueHintType.Time),
+  format: z.string().optional(),
+  ...basePropertyValidators,
+})
+
+const numericDateTimePropertySchema = zodCompleteObject<NumericDateTimeSchema & ObjectSchemaProperty>({
+  type: zodDiscriminant(ValueType.Number),
+  codaType: zodDiscriminant(ValueHintType.DateTime),
+  dateFormat: z.string().optional(),
+  timeFormat: z.string().optional(),
+  ...basePropertyValidators,
+})
+
+const numberPropertySchema = z.union([
+  numericPropertySchema, 
+  scalePropertySchema, 
+  sliderPropertySchema, 
+  currencyPropertySchema,
+  numericDatePropertySchema,
+  numericTimePropertySchema,
+  numericDateTimePropertySchema,
+  baseNumberPropertySchema,
```

Comment:
I don't think you need this, it's overly-permissive for the same reason I mentioned below for baseStringPropertySchema, and you've already covered this case with numericPropertySchema. I removed it in https://github.com/coda/packs-sdk/commit/8249aa6763985634ce2c1d77d0f749bbdb7e0c3f